js/oanda.js

```diff
@@ -0,0 +1,1632 @@
+                result.push (parsed);
+            }
+        }
+        return result;
+    }
+
+    parsePosition (position, market = undefined, symbols = undefined) {
+        // fetchPositions :
+        //
+        //  {
+        //    id: '54',
+        //    instrument: 'EUR_USD',
+        //    price: '1.14531',
+        //    openTime: '2022-02-04T18:47:36.387316038Z',
+        //    initialUnits: '2',
+        //    state: 'OPEN',
+        //    currentUnits: '2',
+        //    realizedPL: '0.0000',
+        //    financing: '0.0000',
+        //    dividendAdjustment: '0.0000',
+        //    unrealizedPL: '-0.0003',
+        //    marginUsed: '0.2290'
+        //  }
+        const marketId = this.safeString (position, 'instrument');
+        market = this.safeMarket (marketId, market);
+        const symbol = market['symbol'];
+        if (symbols !== undefined) {
+            if (symbols.indexOf (symbol) < 0) {
+                return undefined;
+            }
+        }
+        const date = this.safeString (position, 'openTime');
+        const timestamp = this.parseDate (date);
+        const initialSize = this.safeString (position, 'initialUnits');
+        const side = Precise.stringGt (initialSize, '0') ? 'long' : 'short';
+        const marginUsed = this.safeString (position, 'marginUsed');
+        // TODO: i am not sure if the margin values are correctly structured by me
+        return {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'initialMargin': undefined,
+            'initialMarginPercentage': undefined,
+            'maintenanceMargin': undefined,
+            'maintenanceMarginPercentage': undefined,
+            'collateral': marginUsed,
+            'entryPrice': this.safeNumber (position, 'price'),
+            'notional': undefined,
+            'leverage': undefined,
+            'unrealizedPnl': this.safeNumber (position, 'unrealisedPnl'),
+            'contracts': this.parseNumber (Precise.stringAbs (initialSize)),
+            'contractSize': undefined,
+            'realisedPnl': this.safeNumber (position, 'realizedPL'),
+            'marginRatio': undefined,
+            'liquidationPrice': undefined,
+            'markPrice': undefined,
+            'marginType': undefined,
+            'side': side,
+            'percentage': undefined,
+            'status': this.parsePositionStatus (this.safeString (position, 'state')),
+            'info': position,
+        };
+    }
+
+    parsePositionStatus (status, parseBackward = false) {
+        let statuses = {
+            'ALL': 'all',
+            'OPEN': 'open',
+            'CLOSED': 'closed',
+            'CLOSE_WHEN_TRADEABLE': 'unknown',
+        };
+        if (parseBackward) {
+            statuses = this.changeKeyValue (statuses);
+        }
+        return this.safeString (statuses, status, status);
+    }
+
+    async getTransactionsSinceId (sinceId, params) {
```

Comment:
Let's rename this to fetchTransactions and handle `id` in params.

Reply:
this was not 'transactions' in terms of CCXT. it was their concept of 'transactions' which are actually ccxt's 'ledger' concept. I've embedded that code (function) inside another function, where it was used only once.